File: 30daysOFcode_C#/day19.cs
using System;

public interface AdvancedArithmetic{
    int divisorSum(int n);
}

class Calculator : AdvancedArithmetic{
    public int divisorSum(int n){
        int sum = 0;

        for(int i = n; i > 0; i--){
            if(n % i == 0){
                sum += i;
            }
        }

        return sum;

    }
}

class Solution{

    static void Main(string[] args){
        
        int n = Int32.Parse(Console.ReadLine());
      	AdvancedArithmetic myCalculator = new Calculator();
        int sum = myCalculator.divisorSum(n);
        Console.WriteLine("I implemented: AdvancedArithmetic\n" + sum);
    }
}

File: 30daysOFcode_TypeScript/day 19.ts
interface AdvancedArithmetic {
    divisorSum(n: number): number;
}

class Calculator implements AdvancedArithmetic {

    divisorSum(n: number): number {
        let sum: number = 0;

        for (let i: number = 1; i <= n; i++) {
            if (n % i == 0) {
                sum += i;
            }
        }
        return sum;
    }
}